test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/RemoteDependencyWalkerTests.cs

```diff
@@ -1881,6 +1881,177 @@ public async Task TransitiveDependenciesFromNonRootLibraries_AreIgnored()
             AssertPath(d.DowngradedTo, "A 1.0", "D 1.0");
         }
 
+        /// <summary>
+        /// A -> B 1.0.0 -> C 1.0.0 -> D 1.0.0 (this will be rejected)-> E 1.0.0
+        ///
+        ///  D has version defined centrally 2.0.0
+        ///  D 2.0.0 -> I 2.0.0 (this will be downgraded due to central I 1.0.0)
+        ///  (D 2.0.0 should have parentNode C 1.0.0)
+        ///  
+        ///  I has version defined centrally 1.0.0
+        ///  I 1.0.0 -> G 1.0.0
+        ///  (I 1.0.0 should have parentNode D 2.0.0)
+        /// </summary>
+        [Fact]
+        public async Task WalkAsync_WithCentralTransitiveDependency_InnerNodesAndParentNodesCreatedCorrectly()
+        {
+            var framework = NuGetFramework.Parse("net45");
+
+            var context = new TestRemoteWalkContext();
+            var provider = new DependencyProvider();
+            var version100 = "1.0.0";
+            var version200 = "2.0.0";
+
+            // A -> B 1.0.0 -> C 1.0.0(this will be rejected) -> D 1.0.0 -> E 1.0.0
```

Comment:
This is bit different from line 1885.
```suggestion
            // A -> B 1.0.0 -> C 1.0.0 -> D 1.0.0 (this will be rejected)-> E 1.0.0
```

Reply:
Good catch! Fixed.